TheMET/TheMET/Network/ArtFileManager.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ArtFileManager.swift
+//  TheMET
+//
+//  Created by Анна Ситникова on 26/05/2023.
+//
+
+import Foundation
+import UIKit
+
+class ArtFileManager {
+    
+    private let folderURL: URL
+    
+    init?(folderURL: URL) {
+        if folderURL.isFileURL,
+           folderURL.hasDirectoryPath {
+           self.folderURL = folderURL
+       } else {
+           return nil
+       }
+    }
+    
+    func write(art: Art) {
+        let jsonEncoder = JSONEncoder()
+        guard let jsonData = try? jsonEncoder.encode(art) else {
+            return
+        }
+        guard String(data: jsonData, encoding: .utf8) != nil else {
+            return
+        }
+        let fileManager = FileManager.default
+        let fileURL = self.makeFileUrl(artId: art.objectID)
+        DispatchQueue.global().sync {
+            if !fileManager.fileExists(atPath: self.folderURL.path) {
+                do {
+                    try fileManager.createDirectory(at: self.folderURL, withIntermediateDirectories: true)
+                } catch {
+                    return
+                }
+            }
+            fileManager.createFile(atPath: fileURL.path(), contents: jsonData)
+        }
+    }
+    
+    func readArt(id: Int, completion: @escaping (Art?) -> Void) {
+        let fileManager = FileManager.default
+        guard fileManager.fileExists(atPath: self.folderURL.path(percentEncoded: false)) else {
```

Comment:
The folder check is unnecessary as you check the existence of the file a few lines later

Reply:
fixed